soc/SoC_OCV.py

```diff
@@ -25,49 +29,96 @@ def __init__(self):
 		#This is the Resistance present when the chg-discharge curve was done, not the pack resistance
 		#Should be eplicitly tested on the setup. This is a rough estimate of what I expect it was, as we never measured it.
 		IR = 0.085
-		
+
 		#read data from txt files generated during discharge test
 		with open(fName) as f_in:
-			#fill matrices with data from file
-			voltage, current, soc = np.loadtxt(itertools.islice(f_in, 1, None, 2), delimiter = '\t', usecols = (1,3,9), unpack = True)
+			#fill matrices with data from file (all are lists)
+			self.voltage, self.current, self.soc = np.loadtxt(itertools.islice(f_in, 1, None, 2), delimiter = '\t', usecols = (1,3,9), unpack = True)
 		
 		#Estimate the ocv of the cell, only taking the internal resistance into account (assuming no voltage relaxation or recovery)
-		v_ocv = voltage + IR * current
+		self.v_ocv = self.voltage + IR * self.current
+
+		self.soc = self.__selection_sort__([value for value in self.soc])
+		self.v_ocv =  self.__selection_sort__([value for value in self.v_ocv])
 		
 		#generate model for SoC_OCV curve
 		#for now, fitting a 5th degree polynomial, but should use splines with interpolation
-		soc = soc.reshape(-1,1)
-		voltage = voltage.reshape(-1,1)
-		poly_model = make_pipeline(PolynomialFeatures(5), linear_model.Ridge())
-		poly_model.fit(soc, v_ocv)
 		
-		self.voltage_predict = poly_model.predict(soc)
+		#self.voltage = self.voltage.reshape(-1,1)
+		#poly_model = make_pipeline(PolynomialFeatures(5), linear_model.Ridge())
+		#poly_model.fit(self.soc, self.v_ocv)
+		
+		#self.voltage_predict = poly_model.predict(self.soc)
 	
 	#similar to the map function in Arduino
-	def valmap(self, value, istart, istop, ostart, ostop):
-		return ostart + (ostop - ostart) * ((value - istart) / (istop - istart))
+	#def valmap(value, istart, istop, ostart, ostop):
+	#	return ostart + (ostop - ostart) * ((value - istart) / (istop - istart))
 	
 	#SoC between 
-	def get_cell_ocv(self, soc, max = 100):
-		loc = self.valmap(soc, 0, max, len(self.voltage_predict) - 1, 0)
-		if loc > len(self.voltage_predict) - 1:
-			loc = len(self.voltage_predict)-1
-		if loc < 0:
-			loc = 0
-		#return int(loc)
-		return self.voltage_predict[int(loc)]
-		
+	#def get_cell_ocv(self, soc, max = 100):
+	#	loc = self.valmap(soc, 0, max, len(self.voltage_predict) - 1, 0)
+	#	if loc > len(self.voltage_predict) - 1:
+	#		loc = len(self.voltage_predict)-1
+	#	if loc < 0:
+	#		loc = 0
+	#	return int(loc)
+	#	return self.voltage_predict[int(loc)]
+	
+	def get_soc_and_v_ocv(self):
+		return self.soc, self.v_ocv
+	
+	# 
+	def check_in_graph(self, soc2):
+		if (soc2 > self.soc[len(self.soc) - 1] or soc2 < self.soc[0]):
```

Comment:
nitpicky -> `self.soc[len(self.soc) - 1]` has a pretty Python alternative of `self.soc[-1]`

we also don't need brackets around `if` statements 😃